### PR TITLE
Schema Compare cleanup

### PR DESCRIPTION
**Pull request by kisantia**
First pass for cleaning up sqllint errors in a few schema compare files

### CODE REVIEW

File: extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts [hunk 1]
```diff
@@ -183,15 +183,15 @@ export class SchemaCompareOptionsDialog {
 		this.deploymentOptions = defaultOptions;
 	}
 
-	protected async initializeDialog() {
+	protected async initializeDialog(): Promise<void> {
 		this.generalOptionsTab = azdata.window.createTab(loc.GeneralOptionsLabel);
```
Comment (udeeshagautam) [inline extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts:187]: does anything inside this function awaitable?

Comment (kisantia) [reply extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts:187]: oops good catch

File: extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts [hunk 2]
```diff
@@ -264,17 +264,17 @@ export class SchemaCompareOptionsDialog {
...
 				let row = this.optionsTable.selectedRows[0];
 				let label = this.optionsLabels[row];
-				this.descriptionText.updateProperties({
+				await this.descriptionText.updateProperties({
 					value: this.GetDescription(label)
 				});
 			}));
 
-			this.disposableListeners.push(this.optionsTable.onCellAction(async (rowState) => {
+			this.disposableListeners.push(this.optionsTable.onCellAction((rowState) => {
```
Comment (SakshiS-harma) [inline extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts:277]: Why was "async" removed from here?

File: extensions/schema-compare/src/test/testUtils.ts [hunk 3]
```diff
@@ -109,7 +109,7 @@ export async function setDacpacEndpointInfo(path: string): Promise<mssql.SchemaC
 	return endpointInfo;
 }
 
-export async function setDatabaseEndpointInfo(): Promise<mssql.SchemaCompareEndpointInfo> {
+export function setDatabaseEndpointInfo(): mssql.SchemaCompareEndpointInfo {
```
Comment (SakshiS-harma) [inline extensions/schema-compare/src/test/testUtils.ts:112]: There are a few lines in schemaCompare.test.ts that needs to be fixed (await removed) wherever these two functions are called (setDatabaseEndpointInfo, setDacpacEndpointInfo).

File: extensions/schema-compare/src/schemaCompareMainWindow.ts [hunk 4]
```diff
@@ -725,7 +725,7 @@ export class SchemaCompareMainWindow {
 			TelemetryReporter.sendActionEvent(TelemetryViews.SchemaCompareMainWindow, 'SchemaCompareOptionsOpened');
 			// create fresh every time
 			this.schemaCompareOptionDialog = new SchemaCompareOptionsDialog(this.deploymentOptions, this);
-			await this.schemaCompareOptionDialog.openDialog();
+			this.schemaCompareOptionDialog.openDialog();
```
Comment (SakshiS-harma) [inline extensions/schema-compare/src/schemaCompareMainWindow.ts:728]: Just an FYI, I have updated the openDialog method and it is async now- so will be needing an await here. But may be for this PR, go ahead with these changes and I will update it in a separate PR :).

File: extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts (hunk 2 above)
Comment (kisantia) [reply extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts:277]: nothing is being awaited in the callback

File: extensions/schema-compare/src/test/testUtils.ts (hunk 3 above)
Comment (kisantia) [reply extensions/schema-compare/src/test/testUtils.ts:112]: removed those

File: extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts (hunk 2 above)
Comment (SakshiS-harma) [reply extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts:129]: How about line 295, where an "await" is added?

**Pull request merged**